postgis/postgis.sql.in

```diff
@@ -3476,12 +3476,19 @@ CREATE OR REPLACE FUNCTION ST_OffsetCurve(line geometry, distance float8, params
        COST 1; -- reset cost, see #3675
 
 -- Availability: 2.3.0
-CREATE OR REPLACE FUNCTION ST_GeneratePoints(area geometry, npoints numeric)
+CREATE OR REPLACE FUNCTION ST_GeneratePoints(area geometry, npoints integer)
        RETURNS geometry
        AS 'MODULE_PATHNAME','ST_GeneratePoints'
-       LANGUAGE 'c' IMMUTABLE STRICT _PARALLEL
+       LANGUAGE 'c' VOLATILE STRICT _PARALLEL_RESTRICTED
        COST 1; -- reset cost, see #3675
 
+-- Availability: 3.0.0
+CREATE OR REPLACE FUNCTION ST_GeneratePoints(area geometry, npoints integer, seed integer)
+       RETURNS geometry
+       AS 'MODULE_PATHNAME','ST_GeneratePoints'
+       LANGUAGE 'c' IMMUTABLE STRICT _PARALLEL_RESTRICTED
```

Comment:
This version is not depending on any internal state, why not just _PARALLEL?

Reply:
Sure, I'll take this suggestion. Again, a bit out of my depth of knowledge on the parallel features of postgres.